ivy_tests/test_ivy/test_functional/test_core/test_gradients.py

```diff
@@ -272,44 +420,60 @@ def test_layerwise_gradient_descent_update(
 
 
 # lars_update
-@pytest.mark.parametrize(
-    "ws_n_grads_n_lr_n_wsnew",
-    [
-        (
-            Container({"a": [3.0], "b": [3.0]}),
-            Container({"a": [6.0], "b": [6.0]}),
-            Container({"a": [0.1], "b": [0.2]}),
-            Container({"a": [2.7], "b": [2.4]}),
-        )
-    ],
+@given(
+    dtype_and_w=helpers.dtype_and_values(ivy_np.valid_float_dtypes),
+    as_variable=st.booleans(),
+    native_array=st.booleans(),
+    num_positional_args=helpers.num_positional_args(fn_name="lars_update"),
+    container=helpers.list_of_length(st.booleans(), 2),
+    instance_method=st.booleans(),
+    dcdw=st.floats(allow_infinity=False,allow_nan=False),
+    lr=st.floats(min_value=0.0,max_value=1.0,allow_nan=False),
+    decay_lambda=st.floats(min_value=0.0,max_value=1.0,allow_nan=False),
+    inplace=st.booleans(),
+    stop_gradients=st.booleans()
 )
-@pytest.mark.parametrize("dtype", ["float32"])
-@pytest.mark.parametrize("tensor_fn", [ivy.array, helpers.var_fn])
-def test_lars_update(ws_n_grads_n_lr_n_wsnew, dtype, tensor_fn, device, call):
-    # smoke test
-    ws_raw, dcdws_raw, lr_raw, ws_raw_new = ws_n_grads_n_lr_n_wsnew
-    ws = ws_raw.map(lambda x, _: ivy.variable(ivy.array(x)))
-    dcdws = dcdws_raw.map(lambda x, _: ivy.array(x))
-    lr = lr_raw.map(lambda x, _: ivy.array(x))
-    ws_true_new = ws_raw_new.map(lambda x, _: ivy.variable(ivy.array(x)))
-    ws_new = ivy.lars_update(ws, dcdws, lr)
-    # type test
-    assert isinstance(ws_new, dict)
-    # cardinality test
-    for (w_new, w_true_new) in zip(ws_new.values(), ws_true_new.values()):
-        assert w_new.shape == w_true_new.shape
-    # value test
-    for (w_new, w_true_new) in zip(ws_new.values(), ws_true_new.values()):
-        assert np.allclose(ivy.to_numpy(w_new), ivy.to_numpy(w_true_new))
-    # compilation test
-    if call in [helpers.torch_call]:
-        # pytorch scripting does not support internal function definitions
-        return
-
+def test_lars_update(
+    dtype_and_w,
+    as_variable,
+    native_array,
+    num_positional_args,
+    container,
+    instance_method,
+    fw,
+    dcdw,
+    lr,
+    decay_lambda,
+    inplace,
+    stop_gradients
+):
+    dtype, w = dtype_and_w
+    w = np.asarray(w,dtype=dtype)
+    dcdw = np.asarray(dcdw,dtype=dtype)
+    if fw == "torch":
+        w = torch.as_tensor(w)
```

Comment:
why are we converting to `torch.tensor` instances here? I don't think this is necessary.